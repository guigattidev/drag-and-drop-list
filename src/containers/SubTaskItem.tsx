import { useDispatch } from 'react-redux';

import { deleteSubtask, toggleSubtaskDone } from '../redux/todoSlice';
import { ISubTaskItemProps } from '../helpers/app.interfaces';

const SubTaskItem = ({ idTask, idSubTask, content, isDone }: ISubTaskItemProps) => {
  const dispatch = useDispatch();

  const handleDeleteSubtask = () => {
    dispatch(deleteSubtask({ taskId: idTask, subtaskId: idSubTask }));
  };

  const handleToggleSubtaskDone = () => {
    dispatch(toggleSubtaskDone({ taskId: idTask, subtaskId: idSubTask }));
  };

  return (
    <div className={`w-full flex py-3 items-center`}>
      <div onClick={handleDeleteSubtask}>
        <svg
          xmlns="http://www.w3.org/2000/svg"
          fill="none"
          viewBox="0 0 24 24"
          strokeWidth="2"
          stroke="currentColor"
          className={`w-6 h-6 text-gray-500 hover:text-red-500 hover:fill-red-100 cursor-pointer`}
        >
          <path
            strokeLinecap="round"
            strokeLinejoin="round"
            d="M9.75 9.75l4.5 4.5m0-4.5l-4.5 4.5M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
          />
        </svg>
      </div>
      <div onClick={handleToggleSubtaskDone}>
        <svg
          xmlns="http://www.w3.org/2000/svg"
          fill="none"
          viewBox="0 0 24 24"
          strokeWidth="2"
          stroke="currentColor"
          className={`w-6 h-6 text-gray-500 hover:text-green-500 hover:fill-green-100 ${
            isDone && 'text-green-600 fill-green-100'
          } cursor-pointer`}
        >
          <path
            strokeLinecap="round"
            strokeLinejoin="round"
            d="M9 12.75L11.25 15 15 9.75M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
          />
        </svg>
      </div>
      <p className={`ml-2 text-grey-darkest ${isDone && 'line-through'}`}>
        <span className={`font-bold`}>{`Subtask ${idTask}.${idSubTask}: `}</span>
        {content}
      </p>
    </div>
  );
};

export default SubTaskItem;
