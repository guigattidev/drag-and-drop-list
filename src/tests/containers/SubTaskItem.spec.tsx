import renderer from 'react-test-renderer';

import SubTaskItem from '../../containers/SubTaskItem';

jest.mock('react-redux', () => ({
  useDispatch: () => jest.fn(),
}));

describe('SubTaskItem', () => {
  it('renders correctly', () => {
    const props = {
      idTask: 1,
      idSubTask: 1,
      content: 'Test Subtask',
      isDone: false,
    };

    const tree = renderer.create(<SubTaskItem {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
