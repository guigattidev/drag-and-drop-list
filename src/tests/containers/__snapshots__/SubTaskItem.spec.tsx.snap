// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubTaskItem renders correctly 1`] = `
<div
  className="w-full flex py-3 items-center"
>
  <div
    onClick={[Function]}
  >
    <svg
      className="w-6 h-6 text-gray-500 hover:text-red-500 hover:fill-red-100 cursor-pointer"
      fill="none"
      stroke="currentColor"
      strokeWidth="2"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9.75 9.75l4.5 4.5m0-4.5l-4.5 4.5M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </svg>
  </div>
  <div
    onClick={[Function]}
  >
    <svg
      className="w-6 h-6 text-gray-500 hover:text-green-500 hover:fill-green-100 false cursor-pointer"
      fill="none"
      stroke="currentColor"
      strokeWidth="2"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9 12.75L11.25 15 15 9.75M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </svg>
  </div>
  <p
    className="ml-2 text-grey-darkest false"
  >
    <span
      className="font-bold"
    >
      Subtask 1.1: 
    </span>
    Test Subtask
  </p>
</div>
`;
